### PR TITLE
Add 2017-07-17-Span across CoreFx (Round 3)

### PR DESCRIPTION
**Pull request by terrajobst**
@stephentoub @weshaggard @bartonjs  @danmosemsft @mellinoe @ericstj

### CODE REVIEW

File: 2017/07-17-Span across CoreFx/README.md [hunk 1]
```diff
@@ -150,4 +147,70 @@ across CoreFX (with some implementations in CoreCLR/CoreRT).
...
+      `WebSocketReceiveResult`, called `ValueWebSocketReceiveResult`.
+* `HttpClient`
+    - Skipped for now; it's unclear how we want to handle this given this type
+      heavily on `Stream`. Maybe this is addressed by spanifying `Stream`?
+* `HashAlgorithm`
+    - We shouldn't spanify `TryTransformBlock` and `TryTransformBlockFinal`
+      because we only broad back `TransformBlock` and `TransformBlockFinal`
+      back for compat. Customers should use `IncrementalHash` instead.
+* `RSA`
+    - Remove `DecryptValue` and `EncrypValue` -- they always throw
```
Comment (vcsjones) [inline 2017/07-17-Span across CoreFx/README.md]: Typo in `EncrypValue`.

File: 2017/07-17-Span across CoreFx/README.md [hunk 2]
```diff
@@ -150,4 +147,70 @@ across CoreFX (with some implementations in CoreCLR/CoreRT).
...
+    - `flags` should match the existing `Receive` method and thus be
+      `socketFlags`
+    - We should add the overloads that return the `SocketError` as an `out`
+      parameter, looking at telemetry it seems those are popular
+* `SocketTaskExtensions`
+    - Fun fact: for .NET Core, these are calling internal APIs directly on
+      `Socket`
+    - Nothing on `Socket` supports cancellation today; we should investigate
+      whether we can support cancellation in principle and add the APIs now.
+* `SocketASyncEventArgs`
```
Comment (jnm2) [inline 2017/07-17-Span across CoreFx/README.md]: `ASync` should be `Async`

File: 2017/07-17-Span across CoreFx/README.md [hunk 3]
```diff
@@ -150,4 +147,70 @@ across CoreFX (with some implementations in CoreCLR/CoreRT).
...
+    - Currently no support for avoiding allocations when the operation completed
+      synchronously. Proposal is use `ValueTask` and a struct-based alternative
+      `WebSocketReceiveResult`, called `ValueWebSocketReceiveResult`.
+* `HttpClient`
+    - Skipped for now; it's unclear how we want to handle this given this type
+      heavily on `Stream`. Maybe this is addressed by spanifying `Stream`?
+* `HashAlgorithm`
+    - We shouldn't spanify `TryTransformBlock` and `TryTransformBlockFinal`
+      because we only broad back `TransformBlock` and `TransformBlockFinal`
+      back for compat. Customers should use `IncrementalHash` instead.
```
Comment (jnm2) [inline 2017/07-17-Span across CoreFx/README.md]: "only <s>broad</s> brought <s>back</s> `TransformBlock` and `TransformBlockFinal` back"?

File: 2017/07-17-Span across CoreFx/README.md [hunk 4]
```diff
@@ -150,4 +147,70 @@ across CoreFX (with some implementations in CoreCLR/CoreRT).
...
+* `RSA`
+    - Remove `DecryptValue` and `EncrypValue` -- they always throw
+    - `HashData` should be protected, not public
+    - The methods `Encrypt`, `Decrypt`, `HashData`, `SignData`, `SignHash`
+      should follow the `Try` pattern and return `bool `and return the numbers
```
Comment (jnm2) [inline 2017/07-17-Span across CoreFx/README.md]: `` `bool `and `` should be `` `bool` and ``

Comment (jnm2) [inline 2017/07-17-Span across CoreFx/README.md]: "numbers of bytes" should be "number of bytes"

File: 2017/07-17-Span across CoreFx/README.md [hunk 5]
```diff
@@ -150,4 +147,70 @@ across CoreFX (with some implementations in CoreCLR/CoreRT).
...
+      of bytes written as `out` parameter
+* `DSA`
+    - The methods `CreateSignature`, `HashData`, `SignData` should follow the
+      `Try` pattern and return `bool `and return the numbers of bytes written as
```
Comment (jnm2) [inline 2017/07-17-Span across CoreFx/README.md]: Same as https://github.com/dotnet/apireviews/pull/40/files#r129437211

File: 2017/07-17-Span across CoreFx/README.md [hunk 6]
```diff
@@ -150,4 +147,70 @@ across CoreFX (with some implementations in CoreCLR/CoreRT).
...
+      `out` parameter
+    - `HashData` should be protected, not public
+* `ECDsa`
+    - The methods `HashData`, `SignData`, `SignHash` should follow the `Try`
+      pattern and return `bool `and return the numbers of bytes written as `out`
```
Comment (jnm2) [inline 2017/07-17-Span across CoreFx/README.md]: Same as https://github.com/dotnet/apireviews/pull/40/files#r129437211

File: 2017/07-17-Span across CoreFx/README.md [hunk 7]
```diff
@@ -150,4 +147,70 @@ across CoreFX (with some implementations in CoreCLR/CoreRT).
...
+    - Option 2: Add a second interface
+    - Option 3: Add a second interface implement it on our types
+    - Option 4: Wait for default implementations of interfaces
+    - Option 5: Do some internal magic
+    - Consensus go with (1) and wait for a compelling use case and then see
+      whether we can go with 4. Otherwise, a combination of (3) and (4) later
+      seems appealing.
+* Open Issues / Other APIs
+    - Collections / Immutable / Metadata
+    - `ImmuatableArray<T>` is probably the only one that's super interesting
```
Comment (jnm2) [inline 2017/07-17-Span across CoreFx/README.md]: "Immuatable" should be "Immutable"

**Comment by jnm2** [comment]
@terrajobst I apologize, let me know if proofreading is getting on your nerves. It's not primarily why I'm here. =)

**Comment by terrajobst** [comment]
Well, I apologize for all the typos, speaking, listening and typing isn't my strong suit (although I've to admit even when I pay a 100% attention my typo rate is quite high)

File: 2017/07-17-Span across CoreFx/README.md (hunk 4 above)
Comment (jnm2) [inline 2017/07-17-Span across CoreFx/README.md]: It should either be "the **number** of bytes written as **an** `out` parameter" or "the numbers of bytes written as `out` parameter**s**."

**Comment by jnm2** [comment]
No need to apologize for typos, no harm done. I'm as guilty of them as the next person. But welcome to seeing the world through my eyes. 😝

File: 2017/07-17-Span across CoreFx/README.md [hunk 8]
```diff
@@ -150,4 +147,70 @@ across CoreFX (with some implementations in CoreCLR/CoreRT).
...
+      whether we can support cancellation in principle and add the APIs now.
+* `SocketAsyncEventArgs`
+    - When the CoreFxLab representation for multiple buffers is stable, we
+      should probably add an alternative for `BufferList`.
+* `WebSocket`
+    - Currently no support for avoiding allocations when the operation completed
+      synchronously. Proposal is use `ValueTask` and a struct-based alternative
+      `WebSocketReceiveResult`, called `ValueWebSocketReceiveResult`.
+* `HttpClient`
+    - Skipped for now; it's unclear how we want to handle this given this type
```
Comment (stephentoub) [inline 2017/07-17-Span across CoreFx/README.md]: We skipped the GetBytesArray method, but accepted the ReadOnlyBufferContent type.

Comment (terrajobst) [inline 2017/07-17-Span across CoreFx/README.md]: OK, I must have missed that. Will fix.

File: 2017/07-17-Span across CoreFx/README.md (hunk 3 above)
Comment (jnm2) [inline 2017/07-17-Span across CoreFx/README.md]: The other "back" was already there, on the next line.

**Pull request merged**